src/miner.cpp

```diff
@@ -526,6 +527,10 @@ void static BitcoinMiner(const CChainParams& chainparams)
                     // Changing pblock->nTime can change work required on testnet:
                     hashTarget.SetCompact(pblock->nBits);
                 }
+
+                /* If we updated the block above, we also have to update
+                   the auxpow object to match the new block hash.  */
+                CAuxPow::initAuxPow(*pblock);
```

Comment:
Yeah, maybe.  I just tried to change the code in the simplest possible way.  I do not think that this has any practical influence (since the code is only run once in a while, and since the CPU mining code is not meant to be used in production at all), thus I want to keep it as simple and close to the upstream code as possible.